undo_redo.py

```diff
@@ -70,10 +70,15 @@ def set_state(self):
         Set the current state of the row object to what was recorded in the instance variables
         :return: No return value
         """
-        # Copy the copied attribute dictionary over to the object reference
+        # Copy the copied attribute dictionary over to the object reference        
         for key in self.obj_ref.__dict__:
-            self.obj_ref.right_button.setEnabled(True)
-            self.obj_ref.left_button.setEnabled(True)
+            if self.obj_ref.right_button.isEnabled() == True:            
```

Comment:
Suggestion: check if right button is enabled also using OR logic so that the statement is more robust and covers more possible scenarios.